superset/common/query_context.py

```diff
@@ -152,13 +152,39 @@ def get_single_payload(
         self, query_obj: QueryObject, force_cached: Optional[bool] = False,
     ) -> Dict[str, Any]:
         """Return results payload for a single quey"""
-        if self.result_type == utils.ChartDataResultType.QUERY:
+        result_type = query_obj.result_type or self.result_type
```

Comment:
nit: could make sense to document the constructor since, not passing `result_type` assumes it is `FULL` or passed on the `query_obj`